client/client.go

```diff
@@ -1,3 +1,65 @@
+/*
+Package client implements everything required for interacting with a notary repository.
+
+Usage
+
+Use this package by creating a new repository object and calling methods on it.
+
+	package main
+
+	import (
+		"net/http"
+		"os"
+		"path/filepath"
+
+		notary "github.com/docker/notary/client"
+		"github.com/docker/notary/cryptoservice"
+		"github.com/docker/notary/storage"
+		"github.com/docker/notary/trustmanager"
+		"github.com/docker/notary/trustpinning"
+	)
+
+	func main() {
+		rootDir := ".trust"
+		if err := os.MkdirAll(trustDir, 0700); err != nil {
+			panic(err)
+		}
+
+		metadata, err := storage.NewSimpleFileStore(filepath.Join(rootDir, "metadata"), "json")
+		if err != nil {
+			panic(err)
+		}
+
+		remote, err := storage.NewHTTPStore(server.URL, "metadata", "json", "key", http.DefaultTransport)
+		if err != nil {
+			panic(err)
+		}
+
+		cryptoService := cryptoservice.NewCryptoService(trustmanager.NewKeyMemoryStore(demoPassRetriever))
+
+		repo, err := notary.NewNotaryRepository(
+			filepath.Join(trustDir, "repo-data"),
+			"docker.io/library/alpine",
+			"https://notary.docker.io",
+			remote,
+			metadata,
+			trustpinning.TrustPinConfig{},
+			cryptoService,
+			NewMemChangelist(),
+		)
+		if err != nil {
+			panic(err)
+		}
+
+		repo.Update(true)
```

Comment:
it might be better to show `repo.ListTargets()`, since it returns output similar to `notary list`?